test/fft_test.rg

```diff
@@ -1,108 +1,409 @@
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currInt = input[x]
+    format.println("{}, ", currInt)
+  end
+  format.println("]\n")
+end
+
+
+-- Task to print out input or output array with fieldspace float
+__demand(__inline, __leaf)
+task print_array_float_real(input : region(ispace(int1d), float), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  for x in input do
+    var currInt = input[x]
+    format.println("{}, ", currInt)
+  end
+  format.println("]\n")
+end
+
+
+-- Task to print out input or output 2D array with fieldspace complex64
+task print_array_2d_double_complex(input : region(ispace(int2d), complex64), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  format.println("Bounds = {}", input.bounds)
+  for x in input do
+    var currComplex = input[x]
+    format.println("index {}: {} + {}j, ", x, currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output 3D array with fieldspace complex64
+task print_array_3d_double_complex(input : region(ispace(int3d), complex64), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  format.println("Bounds = {}", input.bounds)
+  for x in input do
+    var currComplex = input[x]
+    format.println("index {}: {} + {}j, ", x, currComplex.real, currComplex.imag)
+  end
+  format.println("]\n")
+end
+
+-- Task to print out input or output 3D array with fieldspace double
+task print_array_3d_double_real(input : region(ispace(int3d), double), arrayName: rawstring)
+where reads (input) do
+  format.println("\n{} = [", arrayName)
+  format.println("Bounds = {}", input.bounds)
+  for x in input do
+    var currInt = input[x]
+    format.println("{}, ", currInt)
+  end
+  format.println("]\n")
+end
+
+--task compare_regions_double_complex(r1 : region(ispace(int1d), complex64), r2 : region(ispace(int1d), complex64))
+--where reads (r1, r2) do
+--  var regions_same = true
+--  for x in r1 do
+--    if (r1[x].real == r2[x].real) and (r1[x].imag == r2[x].imag) then
+--      format.println("indices match")
+--    else
+--      format.println("Regions are the not the same: r1 real is {}, r2 real is {}, r1 imag is {}, r2 imag is {} for index {}", r1[x].real, r2[x].real, r1[x].imag, r2[x].imag, x)
+--      regions_same = false
+--    end
+--  end
+--  if regions_same == true then
+--    format.println("Regions are the same")
+--  end
+--  return regions_same
+--end
+
+----SET UP INTERFACES----
+
+--Usage: function fft.generate_fft_interface(itype, dtype): itype = int1d, dtype = complex64, dim = itype.dim =1
+local fft1d = fft.generate_fft_interface(int1d, complex64, complex64)
+local fft2d = fft.generate_fft_interface(int2d, complex64, complex64)
+local fft3d = fft.generate_fft_interface(int3d, complex64, complex64)
```

Comment:
Better to use some consistent naming below so that we can use fewer comments.

e.g. `fft1d_complex32`, `fft2d_float`, etc.

Reply:
Now use the naming convention of `fftXd_<input type>_<output type>`